ReactAndroid/src/main/java/com/facebook/react/modules/network/OkHttpClientProvider.java

```diff
@@ -31,20 +31,32 @@
 
   // Centralized OkHttpClient for all networking requests.
   private static @Nullable OkHttpClient sClient;
+  private static @Nullable OkHttpClient sClientForWebSocket;
 
   public static OkHttpClient getOkHttpClient() {
     if (sClient == null) {
       sClient = createClient();
     }
     return sClient;
   }
-  
+
   // okhttp3 OkHttpClient is immutable
   // This allows app to init an OkHttpClient with custom settings.
   public static void replaceOkHttpClient(OkHttpClient client) {
     sClient = client;
   }
 
+  public static OkHttpClient getOkHttpClientForWebSocket() {
```

Comment:
@greghe do you think it's necessary to avoid multiple instantiation of OkHttpClient?